DuckDuckGo/TabSwitcherViewController.swift

```diff
@@ -349,12 +350,12 @@ extension TabSwitcherViewController: TabViewCellDelegate {
             collectionView.reloadData()
         } else {
             collectionView.performBatchUpdates({
-                isProcessingUpdates = true
+                ignoreProcessingUpdates = true
                 delegate.tabSwitcher(self, didRemoveTab: tab)
                 currentSelection = tabsModel.currentIndex
                 collectionView.deleteItems(at: [IndexPath(row: index, section: 0)])
             }, completion: { _ in
-                self.isProcessingUpdates = false
+                self.ignoreProcessingUpdates = false
```

Comment:
Can this completion be called after `dismiss` if so, this might get set back and defeats the point of this PR

Reply:
I think this is pretty unlikely, but I can add another boolean to separate delete scenario from dismiss scenario

Reply:
Not sure if I understand. If the completion is called **after** `dismiss` then the user would not see the updates in the switcher when removing a tab for example.
The selection and title wouldn't update.

Reply:
This is both about timing and about state changes. 
Timing: Assuming there can be scenario in which completion is called after dismiss, that flag would get set to false allowing (incorrectly) for processing of pending updates.
State: expectation is that once that flag is set in 'dismiss', it should never change.

That's why I'm keen to add one and make flag names as follow:
- isProcessingUpdates (as it was)
- isDismissed

That will still result in negation in guards, but it is explaining what is happening.